### ISSUE TITLE
[Wallet Settings / Manage tokens / Tokens List tab] After clicking some times to the external link buttons, the popup focus doesn't behave as expected

### ISSUE THREAD

**Issue by noeliaSD** [status-im/status-app#12614]
# Bug Report

## Description

Here a video to better understand the behavior:
- After clicking some times the external link, the popup behavior regarding mouse events doesn't work as expected since when hovering items, they are not highlighted as they should be.

## Steps to reproduce

https://github.com/status-im/status-desktop/assets/97019400/16057e42-c1bc-458e-9a34-d81a14b89f74

### Additional Information

- Status desktop version: _bcdb37b0c58e8205081d50cbb2dcc48e7feb4edd_
- Operating System: _MacOS Monterey 12.3.1_

**Comment by caybro** [comment]
Too many overlapping MouseAreas imo :)

**Comment by noeliaSD** [comment]
> Too many overlapping MouseAreas imo :)

Do you mean bc of the design specification or that you could see an alternative in code to reduce `MouseAreas`?

**Comment by caybro** [comment]
> > Too many overlapping MouseAreas imo :)
> 
> Do you mean bc of the design specification or that you could see an alternative in code to reduce `MouseAreas`?

Yeah, talking about the actual code (w/o seeing it :)

**Comment by noeliaSD** [comment]
> > > Too many overlapping MouseAreas imo :)
> > 
> > 
> > Do you mean bc of the design specification or that you could see an alternative in code to reduce `MouseAreas`?
> 
> Yeah, talking about the actual code (w/o seeing it :)

The delegate item is a `Rectangle` that contains a `MouseArea`, covering the item highlight behavior, then inside this `Rectangle` there is a `StatusFlatButton` (containing it's own `MouseArea`) so, 2 nested ones.

**Comment by alexjba** [comment]
@noeliaSD can you take this one and do some profiling? I can't really reproduce it. I've just seen it once

**Comment by caybro** [comment]
> > > > Too many overlapping MouseAreas imo :)
> > > 
> > > 
> > > Do you mean bc of the design specification or that you could see an alternative in code to reduce `MouseAreas`?
> > 
> > 
> > Yeah, talking about the actual code (w/o seeing it :)
> 
> The delegate item is a `Rectangle` that contains a `MouseArea`, covering the item highlight behavior, then inside this `Rectangle` there is a `StatusFlatButton` (containing it's own `MouseArea`) so, 2 nested ones.

Yeah, this is likely the problem; would it help to use a real `ItemDelegate` as the delegate, and using/moving its `Rectangle` as the `background` prop?

**Comment by noeliaSD** [comment]
This is a minor bug introduced during the `Tokens Management` functionality done in this milestone. 

Following priorities of milestone 0.15.5 and team workload, this one is moved to the next iteration.

**Comment by caybro** [comment]
I wasn't able to reproduce this one a single time

**Comment by noeliaSD** [comment]
I can systematically reproduce it always!

**Comment by caybro** [comment]
> I can systematically reproduce it always!

I'll probably give this up to someone who actually can reproduce :/

**Comment by iurimatias** [comment]
moved to 2.29 due to lack of space in this milestone

**Comment by noeliaSD** [comment]
Moved to next milestone due to team capacity and priorities.

**Comment by jrainville** [comment]
Closing as stale and possibly already fixed
